config/webpack/plugin/SetupPlugin.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-param-reassign */
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const path = require('path');
+
+/**
+ * Updates the webpack options with defaults that terra-dev-site requires.
+ */
+class TerraDevSitePlugin {
+  constructor({ publicPath } = {}) {
+    this.publicPath = publicPath;
+  }
+
+  apply(compiler) {
+    // Load the site configuration.
+    const production = compiler.options.mode === 'production';
+    const processPath = process.cwd();
+
+    // OUTPUT
+    compiler.options.output.publicPath = this.publicPath;
+
+    // RESOLVE
+    const devSiteConfigPath = path.resolve(path.join(processPath, 'dev-site-config'));
+    compiler.options.resolve.modules.unshift(devSiteConfigPath);
```

Comment:
Why do we need to add the devSite configuration to the resolve.modules configuration?

Reply:
Yes, the site config is generated to the dev-site-config/build folder, this line lets us include it.
See here:
https://github.com/cerner/terra-dev-site/blob/extend-dev-site/src/TerraDevSite.jsx#L10

I think we need to keep it as is, because we want it to resolve relative to the current process and not relative to the location of the TerraDevSites.jsx file.